string file
---
float32[] states
bool success